app/webpacker/lib/utils/wcif.js

```diff
@@ -22,11 +23,11 @@ export function useSaveWcifAction() {
       options = {},
       onError = alertWcifError,
     ) => {
-      const url = `/api/v0/competitions/${competitionId}/wcif`;
+      const url = patchWcifUrl(competitionId, patchOpts);
 
       save(url, wcifData, onSuccess, options, onError);
     },
-    [save],
+    [save, patchOpts],
```

Comment:
You're passing `patchOpts` as an object (`useSaveWcifAction({ skipSchedule: true })`) which means this will be a fresh object (reference) each render, so this `saveWcif` will re-compute every time, negating the use of `useCallback`. If any code is relying on `saveWcif` never re-computing (i.e. using `saveWcif` in a dependency array somewhere), then this will break that code.

This can be avoided by defining `const wcifSaveOptions = { skipSchedule: true }` outside the component. But it doesn't prevent anyone in the future from passing in an object 'incorrectly.'

(I'm not sure if the default value `{}` is a fresh object each time?)

Making the options a parameter to `save`, instead of the hook itself, might work.

Reply:
Fixed! Should now be less recompute-heavy. And this incidentally also fixes the error with the backend not saving, because the PATCH request was accidentally passing `undefined` in the URL under certain circumstances.

Reply:
Everything looks good to me, and works well as far as I can tell.

I wonder if specifying what _should_ be saved, rather than what should _not_ be saved, might be better though? In the past I have been editing groups in delegate dashboard, and also editing the schedule on the site, and overwritten myself, for example.

Reply:
No wait, adding rooms might still cause conflicts when editing in different browser tabs. I will investigate more tomorrow. But your changes certainly haven't made anything worse.

Reply:
Things that can go wrong with 2 tabs open for venues/schedule respectively:
- Delete a venue/room & save, then edit the schedule in that room & save -> 'invalid json' error
- Copy an existing room (in venue editor) & save, then edit anything in the schedule & save -> new copied room is gone
- Maybe other scenarios

Simple things like editing the fields of venues/rooms while editing the schedule _seem_ to work fine, as far as I can tell.